core/modules/comment/comment.module

```diff
@@ -733,10 +733,28 @@ function comment_node_page_additions(Node $node) {
     $comments_per_page = $node_type->settings['comment_per_page'];
     if ($cids = comment_get_thread($node, $mode, $comments_per_page)) {
       $comments = comment_load_multiple($cids);
-      comment_prepare_thread($comments);
-      $build = comment_view_multiple($comments, $node);
-      $build['pager']['#theme'] = 'pager';
-      $additions['comments'] = $build;
+      // In rare cases $comments is an empty array,
+      // for instance if the only comment entity attached to this node
+      // is broken for some reason.
+      if (!empty($comments)) {
```

Comment:
Very nice improvement over the Drupal patch.

Reply:
Regarding the omitted part:

Nice catch! I wasn't able to reproduce the problem with search indexing yesterday, but it turned out that it's reproducible with changed permissions on viewing comments. I'll add that, too.